src/object.d

```diff
@@ -3722,6 +3722,200 @@ template _arrayOp(Args...)
     alias _arrayOp = arrayOp!Args;
 }
 
+// switch template - replaces code in rt/switch_.d
+int __switch(T, caseLabels...)(T[] condition)
+{
+    static if (caseLabels.length == 0)
+    {
+        return -1;
+    }
+    else static if (caseLabels.length == 1)
+    {
+        return __cmp(condition, caseLabels[0]) == 0 ? 0 : -1;
+    }
+    // To be adjusted after measurements
+    else static if (caseLabels.length < 7)
+    {
+        int r = void;
+        if (condition.length == caseLabels[$ / 2].length)
+        {
+            r = __cmp(condition, caseLabels[$ / 2]);
+            if (r == 0) return cast(int) caseLabels.length / 2;
+        }
+        else
+        {
+            r = ((condition.length > caseLabels[$ / 2].length) << 1) - 1;
```

Comment:
Insert: `// Equivalent to (but faster than) condition.length > caseLabels[$ / 2].length ?  1 : -1`